client/components/reader-infinite-stream/index.js

```diff
@@ -8,72 +8,81 @@ import {
 	CellMeasurerCache,
 	CellMeasurer,
 	InfiniteLoader,
-	AutoSizer,
 } from 'react-virtualized';
 import { debounce, noop, get } from 'lodash';
 
 /**
  * Internal Dependencies
  */
-import ConnectedSubscriptionListItem from 'blocks/reader-subscription-list-item/connected';
+import { recordTracksRailcarRender } from 'reader/stats';
 
-// this component is being cleaned up in: https://github.com/Automattic/wp-calypso/pull/14577
 class ReaderInfiniteStream extends Component {
 	static propTypes = {
 		items: PropTypes.array.isRequired,
+		width: PropTypes.number.isRequired,
+		rowRenderer: PropTypes.func.isRequired,
 		fetchNextPage: PropTypes.func,
-		forceRefresh: PropTypes.any, // forceRefresh can be anything. Whenever we want to force a refresh, it should change
+		hasNextPage: PropTypes.func,
 		windowScrollerRef: PropTypes.func,
-		showLastUpdatedDate: PropTypes.bool,
-		followSource: PropTypes.string,
-		itemType: PropTypes.oneOf( [ 'site', 'post' ] ),
+		extraRenderItemProps: PropTypes.object,
+		minHeight: PropTypes.number,
+		renderEventName: PropTypes.string,
+		passthroughProp: PropTypes.any,
 	};
 
 	static defaultProps = {
 		windowScrollerRef: noop,
 		showLastUpdatedDate: true,
-		itemType: 'site',
+		minHeight: 70,
+		hasNextPage: () => false,
 	};
 
 	heightCache = new CellMeasurerCache( {
 		fixedWidth: true,
-		minHeight: 70,
+		minHeight: this.props.minHeight,
 	} );
 
-	oldSiteRowRenderer = ( { index, key, style, parent } ) => {
-		const site = this.props.items[ index ];
-		const feedUrl = get( site, 'feed_URL' );
-		const feedId = +get( site, 'feed_ID' );
-		const siteId = +get( site, 'blog_ID' );
+	// todo what is eventName / ui_algo?
+	recordTraintrackForRowRender = ( { index, railcar, eventName } ) => {
```

Comment:
the idea here is that we can record traintracks for any arbitrary at the time of render just given a railcar and an eventName (rv supplies the index)